### PR TITLE
infinispan documentation

### PR DESCRIPTION
**Pull request by srinivasa-vasu**

### CODE REVIEW

File: pages/using_cache.md [hunk 1]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+[Infinispan](http://infinispan.org/) is a highly performant caching solution that can work as an in-memory local cache as well as clustered cache. If offers support for multiple cache modes,
+  - [local](http://infinispan.org/docs/stable/user_guide/user_guide.html#local_mode)
+  - [invalidation](http://infinispan.org/docs/stable/user_guide/user_guide.html#invalidation_mode)
+  - [distributed](http://infinispan.org/docs/stable/user_guide/user_guide.html#replicated_mode)
+  - [replicated](http://infinispan.org/docs/stable/user_guide/user_guide.html#distribution_mode).
+
+With JHipster, Infinispan can be used for
+
+- Entity L2 cache
+- HTTP session clustering
```
Comment (jdubois) [inline pages/using_cache.md]: Not any more :-)

File: pages/using_cache.md [hunk 2]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+- Application cache in any of the aforesaid mode by making use of jcache[@CacheResult]/spring[@Cacheable] caching abstractions
+
+Following is the pre-configured default configuration:
+
+- Entities operate in invalidation cache mode
+- Sessions operate in distributed cache mode
```
Comment (jdubois) [inline pages/using_cache.md]: Also not any more

File: pages/using_cache.md [hunk 3]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+- For application specific caching, three caching configurations are pre defined
+  - **local-app-data** for caching data local to the nodes
+  - **dist-app-data** for distributed caching of data across nodes (no of copies determined by the distributed replica count)
+  - **repl-app-data** for replicating data across nodes
+
+Eviction, time-to-live and max-entries for each of the individual operation mode in the cache and the replica count for the distributed mode can be fine tuned using the JHipster [common application properties]({{ site.url }}/common-application-properties/). Fine tune the properties in *jhipster.cache.infinispan* for application specific caching and *spring.jpa.properties.[hibernate.cache.x]* for L2 specific caching.
+
+Though Infinispan 9.0.0.Final GA and later releases added support to run Infinispan embedded caching applications on Kubernetes and OpenShift, hibernate dependency is not yet updated to 9.x releases and hence jHipster Infinispan caching apps won't run on Kubernetes and OpenShift.
```
Comment (jdubois) [inline pages/using_cache.md]: hibernate -> Hibernate
jHipster -> JHipster

**Comment by jdubois** [comment]
You also need to update https://jhipster.github.io/common-application-properties/ with your new configuration properties (I just merged them in jhipster/jhipster )

File: pages/using_cache.md [hunk 4]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
     Member [172.18.0.14]:5701 - 7804d535-86fb-46be-b2a5-d7801dc6a4df
     Member [172.18.0.11]:5701 - 6114ae28-56cd-4840-a575-4d73a6003744
     }
+
+## Caching with Infinispan
+
+[Infinispan](http://infinispan.org/) is a highly performant caching solution that can work as an in-memory local cache as well as clustered cache. If offers support for multiple cache modes,
```
Comment (jdubois) [inline pages/using_cache.md]: "if offers" -> "it offers"

File: pages/using_cache.md [hunk 5]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+  - [local](http://infinispan.org/docs/stable/user_guide/user_guide.html#local_mode)
+  - [invalidation](http://infinispan.org/docs/stable/user_guide/user_guide.html#invalidation_mode)
+  - [distributed](http://infinispan.org/docs/stable/user_guide/user_guide.html#replicated_mode)
+  - [replicated](http://infinispan.org/docs/stable/user_guide/user_guide.html#distribution_mode).
+
+With JHipster, Infinispan can be used for
+
+- Entity L2 cache
```
Comment (jdubois) [inline pages/using_cache.md]: "Hibernate L2 cache" is clearer

File: pages/using_cache.md [hunk 6]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+- Application cache in any of the aforesaid mode by making use of jcache[@CacheResult]/spring[@Cacheable] caching abstractions
+
+Following is the pre-configured default configuration:
+
+- Entities operate in invalidation cache mode
+- For application specific caching, three caching configurations are pre defined
+  - **local-app-data** for caching data local to the nodes
+  - **dist-app-data** for distributed caching of data across nodes (no of copies determined by the distributed replica count)
```
Comment (jdubois) [inline pages/using_cache.md]: "no" -> "number"

File: pages/using_cache.md [hunk 7]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+  - **repl-app-data** for replicating data across nodes
+
+Eviction, time-to-live and max-entries for each of the individual operation mode in the cache and the replica count for the distributed mode can be fine tuned using the JHipster [common application properties]({{ site.url }}/common-application-properties/). Fine tune the properties in *jhipster.cache.infinispan* for application specific caching and *spring.jpa.properties.[hibernate.cache.x]* for L2 specific caching.
```
Comment (jdubois) [inline pages/using_cache.md]: "fine tuned" -> "fine-tuned"

File: pages/using_cache.md [hunk 8]
```diff
@@ -49,3 +49,30 @@ When a new node is added, it will register itself to the service discovery (for
...
+
+If the JHipster Registry is enabled, then the host list will be populated from the registry. If the JHipster Registry is not enabled, host discovery will be based on the default transport settings defined in the 'config-file' packaged within the Jar. Infinispan supports discovery natively for most of the platforms like Kubernets/OpenShift, AWS, Azure and Google
```
Comment (jdubois) [inline pages/using_cache.md]: "Within the Jar" is not clear - within the Infinispan Jar?

**Comment by srinivasa-vasu** [comment]
done.

**Comment by deepu105** [comment]
@jdubois this LGTM, what do you think?

**Pull request merged**